### PR TITLE
add create view validator

### PR DESCRIPTION
**Pull request by tuichenchuxin**
For #17915.

Changes proposed in this pull request:
- add create view sharding validator

### CODE REVIEW

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/ddl/impl/ShardingCreateViewStatementValidator.java [hunk 1]
```diff
@@ -35,11 +45,56 @@ public final class ShardingCreateViewStatementValidator extends ShardingDDLState
...
-        // TODO prevent if has need merge select.
+        Optional<SelectStatement> selectStatement = sqlStatementContext.getSqlStatement().getSelect();
+        if (selectStatement.isPresent()) {
+            TableExtractor extractor = new TableExtractor();
+            extractor.extractTablesFromSelect(selectStatement.get());
+            Collection<SimpleTableSegment> tableSegments = extractor.getRewriteTables();
+            for (SimpleTableSegment each : tableSegments) {
+                String logicTable = each.getTableName().getIdentifier().getValue();
+                if (shardingRule.isShardingTable(logicTable)) {
+                    validateShardingTableAndViewIsBind(shardingRule, sqlStatementContext.getSqlStatement().getView().getTableName().getIdentifier().getValue(), logicTable);
```
Comment (strongduanmu) [inline shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/ddl/impl/ShardingCreateViewStatementValidator.java:56]: Can we rename this method to isBindingTables()? And return boolean type.

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/ddl/impl/ShardingCreateViewStatementValidator.java [hunk 2]
```diff
@@ -35,11 +45,56 @@ public final class ShardingCreateViewStatementValidator extends ShardingDDLState
...
+                }
+            }
+        }
     }
     
     @Override
     public void postValidate(final ShardingRule shardingRule, final SQLStatementContext<CreateViewStatement> sqlStatementContext, final List<Object> parameters,
                              final ShardingSphereDatabase database, final ConfigurationProperties props, final RouteContext routeContext) {
+        Optional<SelectStatement> selectStatement = sqlStatementContext.getSqlStatement().getSelect();
+        selectStatement.ifPresent(statement -> validateNeedsMerge(statement, routeContext));
```
Comment (strongduanmu) [inline shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/ddl/impl/ShardingCreateViewStatementValidator.java:66]: Can we rename this method to isContainsNotSupportedViewStatement()? And return boolean type.

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/ddl/impl/ShardingCreateViewStatementValidator.java [hunk 3]
```diff
@@ -35,11 +45,56 @@ public final class ShardingCreateViewStatementValidator extends ShardingDDLState
...
+            throw new ShardingSphereException("Can not support creating view which has query that needs to be merged!");
+        }
+    }
+    
+    private boolean hasGroupBy(final SelectStatement selectStatement) {
+        return selectStatement.getGroupBy().map(GroupBySegment::getGroupByItems).filter(each -> !each.isEmpty()).isPresent();
+    }
+    
+    private boolean hasAggregation(final SelectStatement selectStatement) {
+        return selectStatement.getProjections().getProjections().stream().anyMatch(each -> each instanceof AggregationProjectionSegment);
```
Comment (strongduanmu) [inline shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/ddl/impl/ShardingCreateViewStatementValidator.java:90]: Please do not use stream api.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/ddl/impl/ShardingCreateViewStatementValidator.java [hunk 4]
```diff
@@ -35,11 +45,56 @@ public final class ShardingCreateViewStatementValidator extends ShardingDDLState
     @Override
     public void preValidate(final ShardingRule shardingRule, final SQLStatementContext<CreateViewStatement> sqlStatementContext,
                             final List<Object> parameters, final ShardingSphereDatabase database) {
-        // TODO prevent if has need merge select.
+        Optional<SelectStatement> selectStatement = sqlStatementContext.getSqlStatement().getSelect();
+        if (selectStatement.isPresent()) {
```
[the remark this answers is not in the document]
Comment (tuichenchuxin) [reply shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/ddl/impl/ShardingCreateViewStatementValidator.java]: Thank you. Done.

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/ddl/impl/ShardingCreateViewStatementValidator.java (hunk 1 above)
Comment (tuichenchuxin) [reply shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/ddl/impl/ShardingCreateViewStatementValidator.java]: Thank you. Done.

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/ddl/impl/ShardingCreateViewStatementValidator.java (hunk 2 above)
Comment (tuichenchuxin) [reply shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/ddl/impl/ShardingCreateViewStatementValidator.java]: Thank you. Done.

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/ddl/impl/ShardingCreateViewStatementValidator.java (hunk 3 above)
Comment (tuichenchuxin) [reply shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/ddl/impl/ShardingCreateViewStatementValidator.java]: Thank you. Done.

**Review by strongduanmu — APPROVED** [review]

**Pull request merged**